source/common/upstream/upstream_impl.cc

```diff
@@ -987,7 +988,18 @@ void MainPrioritySetImpl::updateCrossPriorityHostMap(const HostVector& hosts_add
   }
 
   for (const auto& host : hosts_removed) {
-    mutable_cross_priority_host_map_->erase(addressToString(host->address()));
+    auto existing_host = mutable_cross_priority_host_map_->find(addressToString(host->address()));
```

Comment:
@adisuissa is there a preference to use find() vs [] in the project? https://github.com/abseil/abseil-cpp/blob/5a01d0f77e37493570e35aedaa95c4bcf1673c7c/absl/container/internal/raw_hash_set.h#L3361 find might be O(n) as opposed to O(1) but I saw a few other places in the codebase using find with maps as well

Reply:
Can you refer me to where you saw that the time complexity of find is O(n) and of the operator is O(1)?
I think both should be O(1) average complexity, and O(n) worst-case (due to hash conflicts), but I may be mistaken.

Reply:
nit: const

Reply:
https://github.com/abseil/abseil-cpp/blob/master/absl/container/internal/raw_hash_set.h#L3545 I don't know this code well but also couldn't find any docs that explain this definitively. I will defer to your recommendation. I would be fine with keeping it since other places use find() to access this map this is an established pattern